makefile

```diff
@@ -69,8 +69,14 @@ unixPlatforms := linux_amd64 darwin_amd64 $(if $(STAGING_ONLY),,darwin_arm64 lin
 windowsPlatforms := windows_amd64
 
 
-goos := $(shell $(gobin) env GOOS)
-goarch := $(shell $(gobin) env GOARCH)
+goos := $(GOOS)
+goarch := $(GOARCH)
+ifeq ($(goos),)
+goos := $(shell $(gobin) env GOOS 2> /dev/null)
```

Comment:
Does `/dev/null` exist on non-unix systems?

Reply:
Cygwin is a Unix-like system and Make uses its `/dev/null`, so it should work on Windows tests.

I added MacOS/Windows to the CI tests just to check.